driver/driver.go

```diff
@@ -43,31 +48,40 @@ func (b bigQueryDriver) Open(uri string) (driver.Conn, error) {
 }
 
 func configFromUri(uri string) (*bigQueryConfig, error) {
-
 	if !strings.HasPrefix(uri, "bigquery://") {
 		return nil, fmt.Errorf("invalid prefix, expected bigquery:// got: %s", uri)
 	}
 
 	uri = strings.ToLower(uri)
-	path := strings.TrimPrefix(uri, "bigquery://")
-	fields := strings.Split(path, "/")
-
-	if len(fields) == 3 {
-		return &bigQueryConfig{
-			projectID: fields[0],
-			location:  fields[1],
-			dataSet:   fields[2],
-		}, nil
+	u, err := url.Parse(uri)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse connection string: %q", uri)
+	}
+
+	patternDatasetOnly := "/?*"
+	matchDatasetOnly, err := path.Match(patternDatasetOnly, u.Path)
+	if err != nil {
+		return nil, err
 	}
 
-	if len(fields) == 2 {
-		return &bigQueryConfig{
-			projectID: fields[0],
-			location:  "",
-			dataSet:   fields[1],
-		}, nil
+	patternLocationDataset := "/*/*"
+	matchLocationDataset, err := path.Match(patternLocationDataset, u.Path)
+	if err != nil {
+		return nil, err
 	}
 
-	return nil, fmt.Errorf("invalid connection string : %s", uri)
+	if !matchDatasetOnly && !matchLocationDataset {
+		return nil, fmt.Errorf("invalid connection string: %s", uri)
+	}
+
+	location, dataset := path.Split(u.Path)
+	location = strings.Trim(location, "/")
+	scopes := strings.Split(u.Query().Get("scopes"), ",")
```

Comment:
This generates an array with an empty string when the `scopes` query argument is missing or empty. We should use an empty array instead.